diff -Naur lldb-9.0.0.src.orig/scripts/CMakeLists.txt lldb-9.0.0.src/scripts/CMakeLists.txt
--- lldb-9.0.0.src.orig/scripts/CMakeLists.txt	2019-07-17 00:14:01.000000000 +0900
+++ lldb-9.0.0.src/scripts/CMakeLists.txt	2019-10-20 06:59:58.924277285 +0900
@@ -42,14 +42,28 @@
 )
 
 if(NOT LLDB_BUILD_FRAMEWORK)
+  # Call a python script to gather the arch-specific libdir for
+  # modules like the lldb module.
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/get_relative_lib_dir.py
+    RESULT_VARIABLE get_libdir_status
+    OUTPUT_VARIABLE relative_libdir
+    )
+  if (get_libdir_status EQUAL 0)
+    string(REGEX REPLACE "/python[0-9]+\.[0-9]+/site-packages" ""
+      PYTHON_LIBDIR "${relative_libdir}")
+  else()
+      set(PYTHON_LIBDIR lib${LLVM_LIBDIR_SUFFIX})
+  endif()
+
   if(CMAKE_SYSTEM_NAME MATCHES "Windows")
     set(swig_python_subdir site-packages)
   else()
     set(swig_python_subdir python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
   endif()
 
-  set(SWIG_PYTHON_DIR ${LLVM_LIBRARY_OUTPUT_INTDIR}/${swig_python_subdir})
-  set(SWIG_INSTALL_DIR lib${LLVM_LIBDIR_SUFFIX})
+  set(SWIG_PYTHON_DIR ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${PYTHON_LIBDIR}/${swig_python_subdir})
+  set(SWIG_INSTALL_DIR ${PYTHON_LIBDIR})
 
   # Install the LLDB python module
   install(DIRECTORY ${SWIG_PYTHON_DIR} DESTINATION ${SWIG_INSTALL_DIR})
diff -Naur lldb-9.0.0.src.orig/scripts/get_relative_lib_dir.py lldb-9.0.0.src/scripts/get_relative_lib_dir.py
--- lldb-9.0.0.src.orig/scripts/get_relative_lib_dir.py	2015-10-13 05:12:27.000000000 +0900
+++ lldb-9.0.0.src/scripts/get_relative_lib_dir.py	2019-10-20 06:06:46.778702445 +0900
@@ -23,7 +23,7 @@
     # right answer always.
     arch_specific_libdir = distutils.sysconfig.get_python_lib(True, False)
     split_libdir = arch_specific_libdir.split(os.sep)
-    lib_re = re.compile(r"^lib.+$")
+    lib_re = re.compile(r"^lib.*$")
 
     for i in range(len(split_libdir)):
         match = lib_re.match(split_libdir[i])
